lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -787,6 +800,33 @@ def test_bypass_course_info(self):
         response = views.course_info(request, course_id)
         self.assertEqual(response.status_code, 200)
 
+    def test_accordion(self):
+        table_of_contents = toc_for_course(
+            self.request.user,
+            self.request,
+            self.course,
+            unicode(self.course.get_children()[0].scope_ids.usage_id),
+            None,
+            None
+        )
+
+        # removes newlines and whitespace from the returned view string
+        view = ''.join(render_accordion(self.request, self.course, table_of_contents['chapters']).split())
+        # the course id unicode is re-encoded here because the quote function does not accept unicode
+        course_id = quote(unicode(self.course.id).encode("utf-8"))
+
+        self.assertIn(
+            u'href="/courses/{}/courseware/Chapter_1/Sequential_1/"><p>Sequential1</p>'
+            .format(course_id.decode("utf-8")),
+            view
+        )
+
+        self.assertIn(
+            u'href="/courses/{}/courseware/Chapter_1/Sequential_2/"><p>Sequential2</p>'
+            .format(course_id),
```

Comment:
Good catch! Not intentional. I suppose this is proof that it works either way and the error message yesterday was a red herring but still.